lib/ace/ext/papeeria/latex_behaviour.coffee

```diff
@@ -0,0 +1,187 @@
+    rowDiff = selection.end.row - selection.end.row
+    return {
+      text: opening + selected + closing,
+      selection: [
+        0,
+        selection.start.column + 1,
+        rowDiff,
+        selection.end.column + (if rowDiff != 0 then 0 else 1)
+      ]
+    }
+
+  correspondingClosing = {
+    '(': ')',
+    '[': ']'
+    '{': '}'
+  }
+
+  getBracketInsertionAction = (opening) ->
+    closing = correspondingClosing[opening]
+    return (state, action, editor, session, text) ->
+      { row, column } = editor.getCursorPosition()
+      line = session.getLine(row)
+
+      switch text
+        when opening
+          # Handle non-empty selection case
+          selection = editor.getSelectionRange()
+          selected = session.getTextRange(selection)
+          if selected != ""
+            if editor.getWrapBehavioursEnabled()
+              return getWrapped(selection, selected, opening, closing)
+            else
+              return null
+
+          token = session.getTokenAt(row, column)
+          # Handle escaped bracket case
+          if isEscaped(line, column)
+            shouldComplete = (
+              opening != '{' and
+              not isInEquation(session, row, column)
+            )
+            if shouldComplete
+              return {
+                text: opening + '\\' + closing,
+                selection: [1, 1]
+              }
+            else
+              return null
+
+          # Handle default case
+          if not editor.completer?.activated and not (isCommentToken(token) and column != 0)
+            return { text: opening + closing, selection: [1, 1] }
+
+        when closing
+          nextChar = line[column]
+          # Handle skipping when inserting closing bracket
+          if nextChar == closing
+            matching = session.$findOpeningBracket(closing, { column: column + 1, row: row })
+            if matching?
+              return {
+                text: "",
+                selection: [1, 1]
+              }
+            else
+              return null
+
+          if opening == '{'
+            return null
+
+          # Handle skipping math closing boundary when inserting appropriate bracket
+          if nextChar == "\\" and line[column + 1] == closing and isInEquation(session, row, column)
+            return {
+              text: "",
+              selection: [2, 2]
+            }
+
+  getBracketsDeletionAction = (opening) ->
+    closing = correspondingClosing[opening]
+    return (state, action, editor, session, range) ->
+      if range.isMultiLine()
+        return null
+
+      selected = session.doc.getTextRange(range)
+      if selected != opening
+        return null
+
+      # Handle common bracket deletion case
+      { row, column } = range.start
+      line = session.doc.getLine(row)
+      nextChar = line[column + 1]
+      if nextChar == closing
+        range.end.column += 1
+        return range
+
+      if opening == '{'
```

Comment:
No, it's because they can :) Other brackets, when preceded by `\`, start some sort of math. Escaped `{` is just escaped `{`, so I figured it doesn't need any behavior.